corehq/apps/api/resources/v0_5.py

```diff
@@ -1390,3 +1391,42 @@ def get_datasource_data(request, config_id, domain):
     query = cursor_based_query_for_datasource(request_params, datasource_adapter)
     data = response_for_cursor_based_pagination(request, query, request_params, datasource_adapter)
     return JsonResponse(data)
+
+
+class CommCareAnalyticsUserResource(CouchResourceMixin, HqBaseResource, DomainSpecificResourceMixin):
+
+    roles = fields.ListField()
+    permissions = fields.DictField()
+
+    class Meta(CustomResourceMeta):
+        resource_name = 'analytics-roles'
+        list_allowed_methods = []
+        detail_allowed_methods = ['get']
+
+    def dehydrate_roles(self, bundle):
+        cca_access = get_commcare_analytics_access_for_user_domain(bundle.obj, bundle.request.domain)
```

Comment:
nit: can we avoid calling `get_commcare_analytics_access_for_user_domain` twice in one request?

Reply:
I'll see if I can update to have the `get_obj` handle everything.

Reply:
https://github.com/dimagi/commcare-hq/pull/34987/commits/eea919077eccfaa10c4333b1e54acd6501e03d0d

I simply used the `dehydrate` method to do this instead of the `dehydrate_<field>` methods.